common/configuration/1password_provider.go

```diff
@@ -27,172 +27,154 @@ func (OnePasswordProvider) Role() Secrets                               { return
 	}
 
-	parsedRef, err := decodeSecretRef(string(decoded))
+	// Just to verify that it is JSON encoded.
+	var decoded interface{}
+	err = json.Unmarshal(secret, &decoded)
 	if err != nil {
-		return nil, fmt.Errorf("1Password secret reference invalid: %w", err)
+		return nil, fmt.Errorf("secret is not JSON encoded: %w", err)
 	}
 
-	args := []string{"--format", "json", "item", "get", "--vault", parsedRef.Vault, parsedRef.Item}
-	v, fieldSpecified := parsedRef.Field.Get()
-	if fieldSpecified {
-		args = append(args, "--fields", v)
+	return secret, nil
+}
+
+var vaultRegex = regexp.MustCompile(`^[a-zA-Z0-9_\-.]+$`)
+
+// Store will save the given secret in 1Password via the `op` command.
+//
+// op does not support "create or update" as a single command. Neither does it support specifying an ID on create.
+// Because of this, we need check if the item exists before creating it, and update it if it does.
+func (o OnePasswordProvider) Store(ctx context.Context, ref Ref, value []byte) (*url.URL, error) {
+	if err := checkOpBinary(); err != nil {
+		return nil, err
 	}
-	output, err := exec.Capture(ctx, ".", "op", args...)
-	if err != nil {
-		return nil, fmt.Errorf("run `op` with args %v: %w", args, err)
+	if !vaultRegex.MatchString(o.Vault) {
+		return nil, fmt.Errorf("vault name %q contains invalid characters. a-z A-Z 0-9 _ . - are valid", o.Vault)
 	}
 
-	if fieldSpecified {
-		v, err := decodeSingle(output)
+	url := &url.URL{Scheme: "op", Host: o.Vault}
+
+	_, err := getItem(ctx, o.Vault, ref)
+	if errors.As(err, new(itemNotFoundError)) {
+		err = createItem(ctx, o.Vault, ref, value)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("create item failed: %w", err)
 		}
+		return url, nil
 
-		return json.Marshal(v.Value)
+	} else if err != nil {
+		return nil, fmt.Errorf("get item failed: %w", err)
 	}
 
-	full, err := decodeFull(output)
+	err = editItem(ctx, o.Vault, ref, value)
 	if err != nil {
-		return nil, err
-	}
-
-	// Filter out anything without a value
-	filtered := slices.Filter(full, func(e entry) bool {
-		return e.Value != ""
-	})
-	// Map to id: value
-	var mapped = make(map[string]string, len(filtered))
-	for _, e := range filtered {
-		mapped[e.ID] = e.Value
+		return nil, fmt.Errorf("edit item failed: %w", err)
 	}
 
-	return json.Marshal(mapped)
+	return url, nil
 }
 
-func (o OnePasswordProvider) Store(ctx context.Context, ref Ref, value []byte) (*url.URL, error) {
-	var opref string
-	if err := json.Unmarshal(value, &opref); err != nil {
-		return nil, fmt.Errorf("1Password value must be a JSON string containing a 1Password secret refererence: %w", err)
-	}
-	if !strings.HasPrefix(opref, "op://") {
-		return nil, fmt.Errorf("1Password secret reference must start with \"op://\"")
+func (o OnePasswordProvider) Writer() bool { return o.Vault != "" }
+
+func checkOpBinary() error {
+	_, err := exec.LookPath("op")
+	if err != nil {
+		return fmt.Errorf("1Password CLI tool \"op\" not found: %w", err)
```

Comment:
No need to fix this instance, but convenience fyi for the future - you can denote a string with backticks to avoid having to escape all the quotes inside it:
```
`my "quoted" string`
```